### ISSUE TITLE
feature: i18n routing

### ISSUE THREAD

**Issue by buraksakalli** [aykutkardas/regexlearn.com#60]
Hi Aykut,

I have got an idea about language, languages, and contents should be route-based. This is important for **accessibility** and **SEO**.

Source: https://nextjs.org/docs/advanced-features/i18n-routing

Congrats and thanks for this project! 🥳

**Comment by aykutkardas** [comment]
You're right, I was thinking of doing something about it.

I'll be posting a ROADMAP soon.

Thanks. 🎉

**Comment by aykutkardas** [comment]
Hi @buraksakalli,

RegexLearn uses static generation. That's why it doesn't support i18n. I need to set up a stable structure here. I'd love to know if you have any ideas on this.

**Comment by buraksakalli** [comment]
Hi @aykutkardas,

You need to use `getStaticPaths` for static generation pages.
I have an example project which is a bit complex. 

https://github.com/buraksakalli/nextjs-i18n-country-path-dynamic-routing

You can check;

- next-i18next.config.js
- getI18nPaths.js
- next.config.js
- pages/[country]/electronics/index.js -> example usage

**Comment by aykutkardas** [comment]
Here I have established a dynamic structure that I think is stable and it supports SSG.

If you want to review. @buraksakalli 

Related PR #91 
[pages/_app.js](https://github.com/aykutkardas/regexlearn.com/blob/develop/src/pages/_app.js)
[pages/[lang]](https://github.com/aykutkardas/regexlearn.com/tree/develop/src/pages/%5Blang%5D)
[utils/getIntlPath.js](https://github.com/aykutkardas/regexlearn.com/blob/develop/src/utils/getIntlPath.js)
[components/IntlLink](https://github.com/aykutkardas/regexlearn.com/blob/develop/src/components/IntlLink/index.jsx)
